pkg/registry/generic/etcd/etcd.go

```diff
@@ -285,6 +286,14 @@ func (e *Etcd) Update(ctx api.Context, obj runtime.Object) (runtime.Object, bool
 			if err != nil {
 				return nil, nil, err
 			}
+			if newVersion == 0 {
+				// TODO: The Invalid error should has a field for Resource.
+				// After that field is added, we should fill the Resource and
+				// leave the Kind field empty. See the discussion in #18526.
```

Comment:
I'm fine too - Kind being practically wrong in implementation has made
it useless for the original intent, and since changing error paths is
dangerous we can phase in a new solution for the problem.